server/src/main/java/org/apache/uniffle/server/buffer/AbstractShuffleBuffer.java

```diff
@@ -51,8 +53,16 @@ public abstract class AbstractShuffleBuffer implements ShuffleBuffer {
   public static final long BUFFER_EVICTED = -1L;
 
   public AbstractShuffleBuffer() {
+    this(false);
+  }
+
+  public AbstractShuffleBuffer(boolean enableLAB) {
```

Comment:
Maybe we can extract a class of AbstractLabShuffleBuffer.

Reply:
Yes. We can have multiple inheritance although Java don't support it. We can use the inner class to implement similar function. You google the pattern.